llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -808,98 +796,71 @@ TableFiles processOneModule(std::unique_ptr<Module> M, bool IsEsimd,
   if (IsEsimd && LowerEsimd)
     lowerEsimdConstructs(*M);
 
-  EntryPointGroupMap GMap;
-
-  bool DoSplit = SplitMode.getNumOccurrences() > 0;
-
-  if (DoSplit || DoSymGen) {
-    EntryPointsGroupScope Scope = selectDeviceCodeGroupScope(*M);
-    groupEntryPoints(*M, GMap, Scope);
-  }
+  EntryPointsGroupScope Scope = selectDeviceCodeGroupScope(*M);
```

Comment:
I thought about this simplification.
However, there is a difference in output of  sycl-post-link tool between no split and split in global scope now. Without split option input module is saved almost without modifications (only spec constants update). But if module is split in global scope then only entry points with their dependencies are left and all other device functions are dropped.
I'm ok with setting scope to global if split option is not specified if it won't impact users. But I'd commit this change in a separate PR to leave ability to simply revert that change in case if it will introduce regression.